spacetemplate/parsed_template_blackbox_test.go

```diff
@@ -0,0 +1,267 @@
+package spacetemplate_test
+
+import (
+	"testing"
+
+	"github.com/almighty/almighty-core/resource"
+	"github.com/almighty/almighty-core/spacetemplate"
+	"github.com/almighty/almighty-core/workitem"
+	"github.com/almighty/almighty-core/workitem/link"
+	uuid "github.com/satori/go.uuid"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+// getValidTestTemplate returns a test template in unparsed format. See
+// getValidTestTemplateParsed() for the parsed representation of this template
+func getValidTestTemplate() string {
+	return `
+work_item_types:
+  custom_bug:
+    # planneritem "magically" resolves to 86af5178-9b41-469b-9096-57e5155c3f31 for the witrepo.Create()
+    extends_type: planneritem
+    description: "a basic work item type"
+    name: "Custom Bug"
+    icon: fa fa-bug
+    fields:
+      title:
+        label: "Title"
+        description: "The title of the bug"
+        required: yes
+        type:
+          kind: string
+      state:
+        label: "State"
+        description: "The state of the bug"
+        required: NO
+        type:
+          kind: enum
+          base_type: integer
+          values:
+            - new:
+                label: "New"
+                description: "a new bug"
+                icon: "fa fa-bomb"
+            - closed:
+                label: "Closed"
+                description: "a closed bug"
+                icon: "fa fa-bomb"
+      priority:
+        label: "Priority"
+        description: "The priority of the bug"
+        required: NO
+        type:
+          kind: list
+          component_type: integer
+work_item_link_types:
+  custom_bug_blocker:
+    source_type: custom_bug
+    target_type: custom_bug
+    name: "Custom Bug Blocker"
+    description: "one bug blocks another one"
+    forward_name: "blocks"
+    reverse_name: "blocked by"
+    topology: network
+    # Default link category?
+    #link_category: user
+#cat_mapping:
+#  requirements: # predefined
+#    - bug
+#    - task
+#  foo: # predefined
+#    - bug
+#    - feature
+`
+}
+
+// getValidTestTemplateParsed returns the expected parsed representation of the
+// getValidTestTemplate string
+func getValidTestTemplateParsed() *spacetemplate.ParsedTemplate {
+	bugDesc := "a basic work item type"
+	wiltDesc := "one bug blocks another one"
+	expected := &spacetemplate.ParsedTemplate{
+		WITs: map[string]*spacetemplate.WorkItemTypeHelper{
```

Comment:
The point is to ensure that the YAML above actually gets parsed into this structure, you know?